sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -856,7 +857,31 @@ public <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperati
         return Flux.deferContextual(context -> {
             final BulkExecutor<TContext> executor = new BulkExecutor<>(this, operations, cosmosBulkExecutionOptions);
 
-            return executor.execute().publishOn(CosmosSchedulers.BULK_EXECUTOR_BOUNDED_ELASTIC);
+            return executor
+                .execute()
+                .doFinally((SignalType signal) -> {
+                    if (signal == SignalType.ON_COMPLETE) {
+                        logger.debug("BulkExecutor.execute flux completed - # left items {}, Context: {}",
+                            executor.getItemsLeftSnapshot(),
+                            executor.getOperationContext());
+                    } else {
+                        int itemsLeftSnapshot = executor.getItemsLeftSnapshot();
+                        if (itemsLeftSnapshot > 0) {
```

Comment:
why do we need this if condition? We are anyway just logging this in the next line in both if and else condition.

Reply:
Different log level